docarray/array/array/io.py

```diff
@@ -302,6 +306,75 @@ def to_json(self) -> str:
 
+    @classmethod
+    def from_files(
+        cls,
+        patterns: Union[str, List[str]],
+        recursive: bool = True,
+        size: Optional[int] = None,
+        sampling_rate: Optional[float] = None,
+        read_mode: Optional[str] = None,
+        exclude_regex: Optional[str] = None,
+        *args,
+        **kwargs,
+    ) -> 'DocumentArray':
+        """Creates an iterator over a list of file path or the content of the files.
+        :param patterns: The pattern may contain simple shell-style wildcards, e.g. '\*.py', '[\*.zip, \*.gz]'
+        :param recursive: If recursive is true, the pattern '**' will match any files
+            and zero or more directories and subdirectories
+        :param size: the maximum number of the files
+        :param sampling_rate: the sampling rate between [0, 1]
+        :param read_mode: specifies the mode in which the file is opened.
+            'r' for reading in text mode, 'rb' for reading in binary mode.
+            If `read_mode` is None, will iterate over filenames.
+        :param to_dataturi: if set, then the Document.uri will be filled with DataURI instead of the plan URI
+        :param exclude_regex: if set, then filenames that match to this pattern are not included.
+        :yield: file paths or binary content
+        .. note::
+            This function should not be directly used, use :meth:`Flow.index_files`, :meth:`Flow.search_files` instead
+        """
+        from docarray import DocumentArray
+
+        document_array = []
+        if read_mode not in {'r', 'rb', None}:
+            raise RuntimeError(
+                f'read_mode should be "r", "rb" or None, got `{read_mode}`'
+            )
+
+        def _iter_file_exts(ps):
+            return itertools.chain.from_iterable(
+                glob.iglob(os.path.expanduser(p), recursive=recursive) for p in ps
+            )
+
+        num_docs = 0
+        if isinstance(patterns, str):
+            patterns = [patterns]
+
+        _r = None
+        if exclude_regex:
+            try:
+                _r = re.compile(exclude_regex)
+            except re.error:
+                raise ValueError(f'`{exclude_regex}` is not a valid regex.')
+
+        for g in _iter_file_exts(patterns):
+            if os.path.isdir(g):
+                continue
+            if _r and _r.match(g):
+                continue
+            if sampling_rate is None or random.random() < sampling_rate:
+                if read_mode is None:
+                    d = BaseDocument(uri=g)
+                    document_array.append(d)
+                elif read_mode in {'r', 'rb'}:
+                    with open(g, read_mode) as fp:
+                        d = BaseDocument(content=fp.read(), uri=g)
+                        document_array.append(d)
+                num_docs += 1
+            if size is not None and num_docs >= size:
+                break
+        return DocumentArray[BaseDocument](document_array)
```

Comment:
```suggestion
        return cls(document_array)
```

Here it should return a documentarray of the same cls